.github/scripts/delete-aws-resources.sh

```diff
@@ -72,28 +69,33 @@ function delete_old_aws_resources() {
 
 RETRY_COUNT=0
 while [ $RETRY_COUNT -lt $MAX_RETRIES ]; do
-    # Trigger the function to delete the resources
-    delete_old_aws_resources
-
-    # If the function returned 0, then no resources needed to be deleted
-    # on this run. This means that all resources have been deleted.
-    if [ $? -eq 0 ]; then
-        echo "All resources deleted successfully"
-        break
-    fi
-
-    # Still have resources to delete, increase the retry count
-    RETRY_COUNT=$((RETRY_COUNT + 1))
-
-    # Check if there are more retries left
-    if [ $RETRY_COUNT -lt $MAX_RETRIES ]; then
-        # Retry after delay
-        echo "Retrying in $RETRY_DELAY seconds..."
-        sleep $RETRY_DELAY
-    fi
+  # Trigger the function to delete the resources
+  delete_old_aws_resources
+  status=$?
+
+  # If the function returned 0, then no resources needed to be deleted
+  # on this run. This means that all resources have been deleted.
+  if [ $status -eq 0 ]; then
+    echo "All resources deleted successfully"
+    break
+  fi
+
+  # Still have resources to delete, increase the retry count
+  RETRY_COUNT=$((RETRY_COUNT + 1))
+
+  # Check if there are more retries left
+  if [ $RETRY_COUNT -lt $MAX_RETRIES ]; then
+    # Retry after delay
+    sleep $RETRY_DELAY
+    echo "Retrying in $RETRY_DELAY seconds..."
+
+    # Increase the delay exponentially
+    RETRY_DELAY=$((RETRY_DELAY * 2))
+  fi
 done
 
 # Check if the maximum number of retries exceeded
 if [ $RETRY_COUNT -eq $MAX_RETRIES ]; then
```

Comment:
Is this If clause needed since it is implied when the while at line 71 is exitted?

Reply:
There is an extra **break** case within the while loop: https://github.com/radius-project/radius/pull/7581/files#diff-5625fc3520f54a6e805114f672e280334dac76957f1f5d54cbf3b661f9efeb12R80. That is the success case and if we remove this if `if [ $RETRY_COUNT -eq $MAX_RETRIES ];` then the **echo** command below will always be called.